Updated adapter known problems (#472)
nunit/docs: docs/articles/vs-test-adapter/Known-Problems.md

# Known Problems

> 1. Using both the VSIX adapter and nuget adapters at the same time can lead to discovery issues, ref [Issue769](https://github.com/nunit/nunit3-vs-adapter/issues/769).  The resolution is to uninstall the VSIX adapter, and only use the nuget adapters.  The VSIX adapters are being deprecated in later VS versions.

Reviewer: ```suggestion
1. Using both the VSIX adapter and nuget adapters at the same time can lead to discovery issues; see [nunit3-vs-adapter/issues #769](https://github.com/nunit/nunit3-vs-adapter/issues/769).  The resolution is to uninstall the VSIX adapter, and only use the nuget adapters.  The VSIX adapters are being deprecated in later VS versions.
```

2. Support for Explicit keyword

>     1 Adapter versions in the 3.XX series:  From version 16.2 (Visual Studio 2019) the Explicit keyword is no longer explicitly supported by Visual Studio.  If you want to use it, it does exist as a Category also, so you can use it as a category filter.

Reviewer: ```suggestion
* **Adapter versions in the 3.XX series**:  From version 16.2 (Visual Studio 2019) the `explicit` keyword is no longer explicitly supported by Visual Studio.  If you want to use it, it does exist as a Category also, so you can use it as a category filter.
```

Reviewer 2: Part of the suggested change is incorrect: the NUnit keyword is spelled `Explicit` not `explicit`. Have to beware of what may seem like formatting changes (casing) but are actually content changes.

Reviewer: Thanks @CharliePoole -- this suggestion was made for consistency with another change: https://github.com/nunit/docs/pull/472#discussion_r458043193

It looks like I chose the wrong version to be consistent with. 😄

>     2 Adapter versions in the 4.XX series support the explicit keyword.

Reviewer: ```suggestion
* Adapter versions in the 4.XX series support the `explicit` keyword.
```

> 3. There is no test status in Visual Studio corresponding to NUnit's Inconclusive result, so tests with this result are reported as Not Run. Click on the individual test to see the result.

Reviewer: ```suggestion
3. There is no test status in Visual Studio corresponding to NUnit's `Inconclusive` result, so tests with this result are reported as `Not Run`. Click on the individual test to see the result.
```

Reviewer 2: Nitpick: "Not Run" is not code, it's part of the reported result that is output. I'd use quotes.

4. Theories are reported as individual cases, rather as a single unit.

5. In NUnit, tests have names, which are not necessarily unique. Visual Studio wants the names to be unique. So if two tests have the same name, VS displays a warning message in the output window. The message may be ignored. Two separate results will be shown under the single test in the explorer pane.

6. Startup performance is substantially improved but is still slower than using NUnit directly.

> 7. Applies to pre-VS2017:  A VSIX adapter of older version will be used regardless of version of NuGet adapter.

Reviewer: ```suggestion
7. Applies to pre-VS2017: When using a VSIX adapter and the NuGet adapter, the VSIX adapter will be used regardless of the NuGet adapter version.
```

>    Workaround: Make sure you have upgraded VSIX adapter to latest version, or uninstalled it if you have the NuGet adapter in a solution. The adapter will display its version number in the Output window under Tests.

Reviewer: ```suggestion
Workaround: Make sure you have upgraded VSIX adapter to latest version, or uninstalled it if you have the NuGet adapter in a solution. The adapter will display its version number in the Output window under Tests.
```

   Avoid using the VSIX adapter for VS2017 and upwards.  It is being deprecated, but is still supported optionally in VS2019.

> 8. Visual Studio 2017 Live Unit Testing require NUnit3.  The NUnit2 adapter doesn't support Live Unit Testing.

Reviewer: ```suggestion
8. Visual Studio 2017 Live Unit Testing requires NUnit v3.  The NUnit v2 adapter doesn't support Live Unit Testing.
```

Author: I am not sure about this one, because it is actually NUnit3 adapter and NUnit2 adapter.   E.g.: The NUnit3TestAdapter is now up to version 4......    It was originally version 2 and version 3, but imho it has grown into two different systems.
